src/main/java/org/ibp/api/java/impl/middleware/dataset/DatasetServiceImpl.java

```diff
@@ -388,7 +389,11 @@ public void deleteObservation(
 		this.studyValidator.validate(studyId, true);
 		this.datasetValidator.validateDataset(studyId, datasetId, false);
 		this.observationValidator.validateObservation(studyId, datasetId, observationUnitId, observationId, null);
+
+		final Integer observableId = this.middlewareDatasetService.getPhenotype(observationUnitId, observationId).getObservableId();
+
 		this.middlewareDatasetService.deletePhenotype(observationId);
+		this.middlewareDatasetService.updateDependentPhenotypesAsOutOfSync(observableId, observationUnitId);
```

Comment:
We should abstract updating of dependent phenotypes as out of sync from this service. 
That is, it should be part of the Middleware API call.